cfme/tests/cloud_infra_common/test_policy_simulation.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+"""This module tests events that are invoked by Cloud/Infra VMs."""
+import pytest
+
+from cfme.cloud.provider import CloudProvider
+from cfme.common.vm_views import PolicySimulationDetailsView
+from cfme.common.vm_views import PolicySimulationView
+from cfme.infrastructure.provider import InfraProvider
+from cfme.markers.env_markers.provider import ONE_PER_CATEGORY
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.utils.blockers import BZ
+
+pytestmark = [
+    pytest.mark.usefixtures('setup_provider'),
+    pytest.mark.tier(2),
+    pytest.mark.provider(
+        classes=[CloudProvider, InfraProvider],
+        selector=ONE_PER_CATEGORY,
+        scope="module"
+    ),
+]
+
+
+@pytest.fixture
+def vm_and_filter(provider):
+    collection = provider.appliance.provider_based_collection(provider)
+    vm = collection.all()[0]
+    my_filter = {'name': vm.name, 'provider': provider}
+    yield vm, my_filter
+
+
+@pytest.mark.parametrize("navigation", ["provider_vms", "all_vms", "vm_summary"])
+@pytest.mark.uncollectif(
+    lambda navigation: navigation == "provider_vms" and (BZ(1686617).blocks or BZ(1686619).blocks),
+    reason="This test is blocked by the following BZs: {}, {}".format(
+        BZ(1686617).bug_id,
+        BZ(1686619).bug_id
+    )
+)
+def test_policy_simulation_ui(appliance, vm_and_filter, navigation):
+    """
+    Bugzillas:
+        * 1670456, 1686617, 1686619, 1688359, 1550503
+    Polarion:
+        assignee: jdupuy
+        casecomponent: Control
+        caseimportance: medium
+        initialEstimate: 1/6hr
+    """
+    vm, my_filter = vm_and_filter
+    if navigation != "vm_summary":
+        if navigation == "all_vms":
+            my_filter.pop("provider")
+
+        if vm.provider.one_of(InfraProvider):
+            view = navigate_to(
+                appliance.collections.infra_vms.filter(my_filter),
```

Comment:
I don't see much need for the vm_and_filter fixture - and if you actually move that code to the top of this function, you'll already have the provider-based-collection object that you need to apply the filter to without having this layer of `provider.one_of` checking.  You can also create the filter dictionary based on the navigation parametrization, instead of popping 'provider' off at L53 above.

Reply:
I agree, that's a much cleaner way to do this! :+1: